jquery.xblock.js

```diff
@@ -348,4 +377,19 @@
         });
     };
 
+    // Find all the children of an element that match the selector, but only
+    // the first instance found down any path.  For example, we'll find all
+    // the ".xblock" elements below us, but not the ones that are themselves
+    // contained somewhere inside ".xblock" elements.
+    // Code borrowed from edx-platform and edx-sdk
+    $.fn.highestDescendants = function(selector) {
+        return this.children().map(function(idx, element) {
+            if ($(element).is(selector)) {
+                return element;
+            } else {
+                return $(element).highestDescendants(selector).toArray();
+            }
```

Comment:
@bradenmacdonald inconsistent return type: it returns either single element or an array of elements. Might make sense to unify somehow (e.g. always returning array than flattening it at the top level call)